File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.py
import sys
input = sys.stdin.readline

S = [x for x in input().rstrip()] + ["0"]
alpha = ["c=", 'c-', "d-", "lj", "nj", "s=", "z="]

i = 0
j = 1
cnt = 0

while j < len(S):

    if S[i] == "d" and S[j] == 'z':
        j += 1

        if "".join(S[i:j+1]) == "dz=":
            cnt += 1
            i += 3
            j += 2
            continue
        else:
            cnt += 1
            i += 1
            continue

    find = S[i] + S[j]
    if find in alpha:
        cnt += 1
        i += 2
        j += 2
    else:
        cnt += 1
        i += 1
        j += 1

print(cnt)

File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim()
// let input = fs.readFileSync("./EX.txt").toString().trim()
alpha = ["c=", 'c-', "dz=", "d-", "lj", "nj", "s=", "z="]

for (i of alpha) {
    input = input.replaceAll(i, "*") 
}
console.log(input.length)